### DISCUSSION TITLE
Characters as NPCs or copy them between scenes

### DISCUSSION THREAD

**Discussion by FlamerShaftglutton** [fariapp/fari-app#157] [Ideas]
I'm really liking Fari so far, but I have three closely related frustrations:
1. If I want an NPC in more than one scene I need to manually recreate them in each.
2. If an NPC is coming into a scene later than expected (because of the outcome of a previous scene) I have to delete the NPC, then recreate them later.
3. If I want multiple copies of an NPC (like 3 identical henchmen) I have to manually create them.

I feel like this could be solved by either allowing for Character sheets to be placed in a scene as an NPC, or if NPCs could be copied between scenes. Alternatively if a character or aspect can be hidden from the players then this could be hacked by recreating the NPC in every scene but hiding it until it's needed.

Is there already a way to do this that I'm missing? I've set them up as a Character and put them in the player list, but that's not intuitive since it's separated from the other NPCs who live in the scene.

**Comment by RPDeshaies** [comment]
You are correct. Since "NPCs" are basically just index cards (but green), they are **not** sticky for you entire session.

They are flushed when a scene is reinitialized.

In terms of solutions, you can, like you suggested, use a full on character sheet and import it in the scene.

FYI in the latest release where I added a new section to display index cards representing players so perhaps that alleviate the problem a bit but it's still not super intuitive.

<img src="https://user-images.githubusercontent.com/6224111/99466642-c5361600-290a-11eb-878e-91af58e2c9f1.png" width="400px"/>

But I think a better solution would be to have "pinned" aspects that are sticky for an entire session.

I could use a similar design as Google Keep for example

![image](https://user-images.githubusercontent.com/6224111/99466916-3ece0400-290b-11eb-88a3-f60e49fa0493.png)

**Comment by RPDeshaies** [comment]
What do you think of something like this:

![gif](https://gyazo.com/634ce4500fd60164a3a98e56ac539ce3.gif)

**Reply by FlamerShaftglutton** [reply]
If that works when loading a different scene then that would help with issue number 1. 

Even if it only works for resetting I can see a lot of value in that, though.

**Reply by RPDeshaies** [reply]
you are totally right, I'll make sure it works for loading different scenes as well. Good catch

**Reply by RPDeshaies** [reply]
Will be deployed with https://github.com/fariapp/fari/pull/156

**Comment by pooya-raz** [comment]
This would be really helpful, especially if we can get clickable skills, same way that PCs have.

**Reply by RPDeshaies** [reply]
Pinned Cards are already live on fari.app so you can test it out right now.

Having NPCs with clickable skills could be an interesting addition though... but I don't really know what it would look like "interface wise"
